app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenter.java

```diff
@@ -35,18 +45,24 @@ private void requestContestDetails() {
         Disposable apiCallDisposable = restApi.getContestDetails(contestId)
                 .compose(subscribeOnIoObserveOnUi())
                 .subscribe(response -> {
-                    Contest contest = response.contest;
-                    updateViewWithContest(contest);
+                    updateViewWithContest(response.contest);
                 }, throwable -> {
                     Timber.d("API error retrieving contest details: " + throwable.getMessage());
                     view.showMessage(R.string.error_api);
+                    view.showRatingGuide(createScoreWeights());
                 });
         disposables.add(apiCallDisposable);
     }
 
     private void updateViewWithContest(Contest contest){
         view.showContestName(contest.getTitle());
         view.showContestDescription(contest.getDescription());
+        showCategoriesForContest(contest);
+    }
+
+    private void showCategoriesForContest(Contest contest) {
+        //Dual Adapter requrires categories are displayed before the rating guides.
```

Comment:
Are you saying that calling `categoryAdapter.setWeights(____)` before setting categories would throw an error?
Consider creating one setter for both then, which would handle the order. Otherwise this is not very clear.